snuba_sdk/metrics_query.py

```diff
@@ -92,9 +93,18 @@ def print(self) -> str:
 
     def serialize(self) -> str | dict[str, Any]:
         self.validate()
+        self._optimize()
         result = MQL_PRINTER.visit(self)
         return result
 
+    def _optimize(self) -> None:
+        if isinstance(self.query, str):
```

Comment:
Instead of throwing an exception here, I would simply skip this if the value is a str. It's possible for people to construct queries with raw MQL strings, but we parse them and re-encode them before sending to Snuba.

Reply:
Ok I will make this change.

Another thought I had was if its a string you can parse first and then optimize, but this might make it slower and perhaps cause unexpected side-effects? Do you have any thoughts on this.